include/capypdf.hpp

```diff
@@ -169,6 +176,157 @@ class GraphicsState : public CapyC<CapyPDF_GraphicsState> {
     void set_TK(int32_t value) { CAPY_CPP_CHECK(capy_graphics_state_set_TK(*this, value)); }
 };
 
+class Function : public CapyC<CapyPDF_Function> {
+public:
+    friend class Generator;
+
+    Function(double *domain, int32_t num_domain, Color &c1, Color &c2, double n = 1.0) {
+        CapyPDF_Function *fn;
+        CAPY_CPP_CHECK(capy_type2_function_new(domain, num_domain, c1, c2, n, &fn))
+        _d.reset(fn);
+    }
+
+    Function(double *domain,
+             int32_t num_domain,
+             CapyPDF_FunctionId *functions,
+             int32_t num_functions,
+             double *bounds,
+             int32_t num_bounds,
+             double *encode,
+             int32_t num_encode) {
+        CapyPDF_Function *fn;
```

Comment:
This is not great from a user experience perspective. The only way to tell what `Function x{1, 2, 3, 4}` does is to count the arguments and then work out which constructor that is and what it creates.

There should be a  `Function2`, `Function3` etc as separate types. Basically the same as what Shadings do (with the same review comments).